### ISSUE TITLE
Plot diagram in jupyter notebook

### ISSUE THREAD

**Issue by AntonYurchenko** [haifengl/smile#418]
Hi! 
I use jupyter notebook with BeakerX.
How can i plot diagram in the notebook? It is possible or not?
Right now i catch the exception
```bash
java.awt.HeadlessException:
No X11 DISPLAY variable was set, but this program performed an operation which requires it.
  at java.awt.GraphicsEnvironment.checkHeadless(GraphicsEnvironment.java:204)
  at java.awt.Window.<init>(Window.java:536)
  at java.awt.Frame.<init>(Frame.java:420)
  at javax.swing.JFrame.<init>(JFrame.java:233)
  at smile.plot.Window$.frame(Window.scala:49)
  at smile.plot.Window$.apply(Window.scala:33)
  at smile.plot.Operators.plot(Operators.scala:86)
  at smile.plot.Operators.plot$(Operators.scala:84)
  at smile.plot.package$.plot(package.scala:23)
  ... 94 elided
```
My code in jupyter
```scala
import scala.util.Random
val vectors = (1 to 100).map(_ => (1 to 3).map(_ => Random.nextDouble()).toArray).toArray
val clusters = kmeans(vectors, k = 20, runs = 20)
val labels = clusters.getClusterLabel()
plot(vectors, labels, '.', Palette.COLORS)
```
Thank you!

**Comment by haifengl** [comment]
Smile plot is based on Swing, which I don't think BeakerX supports.

**Comment by AntonYurchenko** [comment]
Please don't close the issue in duration a week. I am working under solution this question. If I will be able to find solution then I describe it here.
Thank you!

**Comment by AntonYurchenko** [comment]
I has found solution of visualisation results of calculation.

**First:** The docker image with Jupyter and BeakerX from official site (`beakerx/beakerx`) has old version BeakerX 1.0.0. In official git repository BeakerX right now is available version 1.4.1. I has built to myself docker image with last version BeakerX. This image is available here https://hub.docker.com/r/entony/jupyter-scala, Dockerfile is available here https://github.com/AntonYurchenko/docker/tree/master/jupyter-scala.

**Second:** BeakerX has tools for visualisation of results. You can find information about the tools in documentation from official git repository of the project here https://github.com/twosigma/beakerx/blob/master/doc/scala/Plot.ipynb

**Third:** Here I give an example visualisation results of k-means clustering for Jupyter with BeakerX.
```
%%classpath add mvn
com.github.haifengl smile-scala_2.11 1.5.2
```
```scala
import smile.clustering._
import scala.util.Random
val vectors = (1 to 100).map(_ => (1 to 2).map(_ => Random.nextDouble()).toArray).toArray
val clusters = kmeans(vectors, k = 3, runs = 20)
```
```scala
val clusteringPoints = (vectors zip clusters.getClusterLabel).groupBy(_._2)
val redPoints = clusteringPoints(0).map(_._1)
val greenPoints = clusteringPoints(1).map(_._1)
val bluePoints = clusteringPoints(2).map(_._1)
```
```scala
val plot = new Plot { title = "Clustering of random points" }

val points = List(
    
    // red points
    new Points {
        x = redPoints.map(_(0))
        y = redPoints.map(_(1))
        size = 5.0
        color = Color.red
    },
    
    // green points
    new Points {
        x = greenPoints.map(_(0))
        y = greenPoints.map(_(1))
        size = 5.0
        color = Color.green
    },
    
    // blue points
    new Points {
        x = bluePoints.map(_(0))
        y = bluePoints.map(_(1))
        size = 5.0
        color = Color.blue
    }
)

plot.add(points)
```
Source: https://github.com/AntonYurchenko/docker/blob/master/jupyter-scala/examples/scala-smile-k-means-example.ipynb
